roles/istio/tasks/certificates.yaml

```diff
@@ -1,24 +1,8 @@
 ---
-- name: Register ssl config options
-  set_fact:
-    server_certs_paths_defined: "{{ istio_ssl.server_crt_path is defined and istio_ssl.server_key_path is defined }}"
-    self_signed_ca: "{{ istio_ssl.self_signed_ca }}"
-
-- name: Check istio_ssl configs
+- name: Check istio_ssl configuration
   fail:
-    msg: >
-      Bad istio_ssl configs: Either server cert and key paths are provided (server_crt_path and server_key_path),
-      or only self_signed_certs must be true: {{ istio_ssl }}
-  when: >
-    (not server_certs_paths_defined and not self_signed_ca) or
-    (server_certs_paths_defined and self_signed_ca)
-
-
-- name: Create temporary certificates directory
-  ansible.builtin.tempfile:
-    state: directory
-    suffix: certificate
-  register: tmp_cert_dir
+    msg: "When not using self signed certificates the certificates paths must be provided."
+  when: not (istio_ssl.self_signed_ca | bool) and (istio_ssl.server_crt_path is undefined or istio_ssl.server_key_path is undefined or istio_ssl.server_crt_path is none or istio_ssl.server_key_path is none)
```

Comment:
What about the situation where someone sets both `istio_ssl.self_signed_ca: true` AND `istio_ssl.server_crt_path` and `istio_ssl.server_key_path` are set

One needs to take precedence over the other? (I think I originally I checked the current one you've got and this other one)

Reply:
I would see `istio_ssl.self_signed_ca` to take precedence. If it is set to `true` then we generate self-signed certs and use them. If set to `false` then we check if paths are provided.